internal/scope/machine.go

```diff
@@ -282,6 +282,39 @@ func (m *MachineScope) GetSSHPublicKey() string {
 	return ""
 }
 
+// GetBasicAuthToken will fetch the BasicAuthSecret on the MvmCluster and
+// and return the token for the given host.
+// If no secret or no value is found, an empty string is returned.
+func (m *MachineScope) GetBasicAuthToken(addr string) string {
+	placement := m.MvmCluster.Spec.Placement
+	if placement.StaticPool == nil || placement.StaticPool.BasicAuthSecret == "" {
+		return ""
+	}
+
+	tokenSecret := &corev1.Secret{}
+	key := types.NamespacedName{
+		Name:      m.MvmCluster.Spec.Placement.StaticPool.BasicAuthSecret,
+		Namespace: m.MvmCluster.Namespace,
+	}
+
+	if err := m.client.Get(m.ctx, key, tokenSecret); err != nil {
+		m.V(2).Info("could not get basicAuthSecret", "error", err) //nolint:gomnd // this magic number is fine
+
+		return ""
+	}
+
+	// If it's not there, that's fine; we will return an empty string
+	token := string(tokenSecret.Data[strings.Split(addr, ":")[0]])
+
+	if token == "" {
+		m.V(2).Info( //nolint:gomnd // this magic number is fine
```

Comment:
If they've supplied a secret name but there is no token do you think we should report an error?

Reply:
you mean error out?
the secret holds tokens for all hosts, so in the case that 1 of the flintlocks does not have auth on it, it would not have an entry in the secret, so erroring would be annoying
i think better logging would be good here, like say which host we have found tokens for and which ones we haven't